test/test_opsum_to_ttn.jl

```diff
@@ -257,6 +256,10 @@ end
         Tmpo_lr = contract(Hsvd_lr)
       end
       @test Tttno_lr ≈ Tmpo_lr rtol = 1e-6
+      # ToDo: move to a more appropriate file, since this is not related to OpSum to TTN conversion
+      # test whether Hamiltonian contracts with itself (used to error due to missing siteindices)
+      @test isa(inner(Hsvd, Hsvd), Number)
```

Comment:
Agreed. Also, it would be good to add tests for `hascommoninds` and `check_hascommoninds` with missing site indices.